### PR TITLE
Better voting error messages

### PR DESCRIPTION
**Pull request by FSadrieh**

### CODE REVIEW

File: evap/staff/templates/staff_semester_view.html [hunk 1]
```diff
@@ -573,6 +578,15 @@ <h3>
             document.getElementById("form_activation_status").requestSubmit();
         }
     </script>
+    {% trans 'Create exam evaluation' as title %}
+    {% blocktrans asvar question %}Create an exam evaluation based on this evaluation. This will copy all the participants and contributors from the original evaluation. It will set the weight of the original evaluation to 9 and its end date will be set to the day before the exam. {% endblocktrans %}
+    {% trans 'Create exam evaluation' as action_text %}
+    {% include 'confirmation_modal.html' with modal_id='createExamEvaluation' title=title question=question action_text=action_text btn_type='danger' %}
+    <script type="text/javascript">
+        function createExamEvaluationAction() {
+            pass
```
Comment (richardebeling) [inline evap/staff/templates/staff_semester_view.html:587]: Seems like changes from another PR are included here, but you marked this PR as review ready -- do you expect us to review already or can I mark as draft here?

File: evap/student/views.py [hunk 2]
```diff
@@ -162,9 +162,12 @@ def render_vote_page(request, evaluation, preview, for_rendering_in_modal=False)
         evaluation_form_group_bottom = []
 
     template_data = {
-        "errors_exist": any(
+        "contributor_errors_exist": any(
+            any(form.errors for form in form_group) for form_group in [*(form_groups.values())]
+        ),
```
Comment (richardebeling) [inline evap/student/views.py:167]: Why `[*(form_groups.values())]`? Could this just be
```suggestion
        "contributor_errors_exist": any(form.errors for form_group in form_groups.values() for form in form_group)
```
?

File: evap/student/templates/student_vote.html [hunk 3]
```diff
@@ -27,6 +27,19 @@
             </div>
         </div>
     {% endif %}
+    {% if general_errors_exist and not contributor_errors_exist %}
```
Comment (richardebeling) [inline evap/student/templates/student_vote.html:30]: The whole outer div is duplicated, the only thing that changed is the inner error message, right? In that case, can you please reduce the scope of the `if` to just the error message (-> deduplicate the outer div, just have the `if` around the `blocktrans` tag)?

File: evap/student/templates/student_vote.html [hunk 4]
```diff
@@ -98,14 +111,14 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
                         <div class="card collapsible{% if not forloop.last %} mb-3{% endif %}">
                             <div class="card-header d-flex tab-row">
                                 <div class="me-auto">
-                                    <button class="collapse-toggle{% if errors_exist and not contributor_has_errors %} collapsed tab-selectable{% endif %} bg-transparent" data-bs-toggle="collapse"
+                                    <button class="collapse-toggle{% if general_errors_exist or contributor_errors_exist and not contributor_has_errors %} collapsed tab-selectable{% endif %} bg-transparent" data-bs-toggle="collapse"
```
Comment (richardebeling) [inline evap/student/templates/student_vote.html:114]: Django templating language follows the common operator precedence of "and" binding stronger than "or" ([docs](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#boolean-operators)). You need parentheses around the `or` here.

File: evap/student/templates/student_vote.html [hunk 5]
```diff
@@ -98,14 +111,14 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
...
                                         data-bs-target="#vote-area-{{ contributor.id }}" aria-expanded="false" aria-controls="vote-area-{{ contributor.id }}" tabindex="-1" type="button">
                                         {{ contributor.full_name }}
                                         {% if label %} &ndash; <span class="fst-italic">{{ label }}</span>{% endif %}
                                     </button>
                                 </div>
                             </div>
-                            <div class="collapse{% if not errors_exist or contributor_has_errors %} show{% endif %}" id="vote-area-{{ contributor.id }}">
+                            <div class="collapse{% if not general_errors_exist and not contributor_errors_exist or contributor_has_errors %} show{% endif %}" id="vote-area-{{ contributor.id }}">
```
Comment (richardebeling) [inline evap/student/templates/student_vote.html:121]: I think the template would be more readable if the view still provided a `errors_exist = general_errors_exist or contributor_errors_exist` variable in the template context -- parsing and understanding the condition here takes a while for me.

File: evap/student/templates/student_vote.html [hunk 6]
```diff
@@ -15,18 +15,23 @@
 
 {% block content %}
     {{ block.super }}
-    {% if errors_exist %}
+    {% if contributor_errors_exist or general_errors_exist %}
```
[the remark this answers is not in the document]
Comment (FSadrieh) [reply evap/student/templates/student_vote.html:18]: This does not work as it does not render the error message at the top of the page, but above the block where the error occurs

File: evap/student/views.py [hunk 7]
```diff
@@ -161,11 +161,16 @@ def render_vote_page(request, evaluation, preview, for_rendering_in_modal=False)
...
+    )
+
     template_data = {
-        "errors_exist": any(
-            any(form.errors for form in form_group)
-            for form_group in [*(form_groups.values()), evaluation_form_group_top, evaluation_form_group_bottom]
-        ),
+        "contributor_errors_exist": contributor_errors_exist,
+        "general_errors_exist": general_errors_exist,
+        "errors_exists": contributor_errors_exist or general_errors_exist,
```
Comment (janno42) [inline evap/student/views.py:173]:
```suggestion
        "errors_exist": contributor_errors_exist or general_errors_exist,
```
maybe this solves your problem above

**Review by richardebeling — APPROVED** [review]

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

**Pull request merged**